packages/psss-server/src/routes/Route.ts

```diff
@@ -3,28 +3,24 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { Request, Response } from 'express';
+import { NextFunction, Request, Response } from 'express';
 import { Route as BaseRoute, ExpressRequest, ExpressResponse } from '@tupaia/server-boilerplate';
-import { UnauthenticatedError } from '@tupaia/utils';
 import { EntityConnection, CentralConnection, ReportConnection } from '../connections';
 
 export class Route<
   Req extends ExpressRequest<Req> = Request,
-  Res extends ExpressResponse<Req> = Response
+  Res extends ExpressResponse<Req> = Response,
 > extends BaseRoute<Req, Res> {
-  protected entityConnection?: EntityConnection;
-  protected centralConnection?: CentralConnection;
-  protected reportConnection?: ReportConnection;
+  protected readonly entityConnection: EntityConnection;
+  protected readonly centralConnection: CentralConnection;
+  protected readonly reportConnection: ReportConnection;
 
-  public handle() {
-    try {
-      const { session } = this.req;
-      this.entityConnection = new EntityConnection(session);
-      this.centralConnection = new CentralConnection(session);
-      this.reportConnection = new ReportConnection(session);
-    } catch (e) {
-      throw new UnauthenticatedError('Unauthenticated');
-    }
-    return super.handle();
+  public constructor(req: Req, res: Res, next: NextFunction) {
+    super(req, res, next);
+    const { session } = this.req;
+    const { entity: entityApi, central: centralApi, report: reportApi } = this.req.ctx.services;
+    this.entityConnection = new EntityConnection(entityApi);
+    this.centralConnection = new CentralConnection(centralApi, session.email);
```

Comment:
Maybe a stupid question, but just making sure if there is no session found this won't throw an error trying to access `email` from `undefined`?

Reply:
I think it should always be defined in this case, since PSSS doesn't support public users, but I'll make it `session?.email` just in case :+1: